source/common/http/http1/conn_pool.h

```diff
@@ -117,6 +119,7 @@ class ConnPoolImpl : public ConnectionPool::Instance, public ConnPoolImplBase {
   std::list<ActiveClientPtr> ready_clients_;
   std::list<ActiveClientPtr> busy_clients_;
   std::list<DrainedCb> drained_callbacks_;
+  std::list<IdleCb> idle_callbacks_;
```

Comment:
it might be overkill, did you consider using `Common::CallbackManager`?

Reply:
Makes sense. I replaced idle_callbacks_ with that. I considered replacing drained_callbacks_, but it has some logic dependent on whether the list is empty, and that wasn't exposed by `common::CallbackManager`. Can probably be added in another PR.